src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx

```diff
@@ -543,66 +568,48 @@ function LineplotViz(props: VisualizationProps) {
-    const yMinMaxRange = categoricalMode
-      ? { min: 0, max: 1 }
-      : data.value?.yMin != null && data.value?.yMax != null
-      ? ({
-          // make sure axis range includes a zero (except for dates, see below)
-          min: min([data.value.yMin, 0]), // lodash min/max handles strings
-          max: max([data.value.yMax, 0]), // or numbers appropriately
-          // and with strings/dates, if you put the zero as the second value, min or max always returns the string
-          // so min(["foo", 0]) is "foo" and min([0, "foo"]) is 0 **
-          // which is our desired behaviour (no need to anchor date axes at 1970 or whatever)
-          // (** TO DO: is this documented lodash behaviour, or likely to change?)
-        } as NumberOrDateRange)
-      : undefined;
+  const defaultIndependentAxisRange = useDefaultAxisRange(
+    xAxisVariable,
+    data.value?.xMin,
+    data.value?.xMinPos,
+    data.value?.xMax,
+    vizConfig.independentAxisLogScale
+  );
 
-    return ((yAxisVariable?.type === 'number' ||
-      yAxisVariable?.type === 'integer' ||
-      (yAxisVariable?.type === 'string' && categoricalMode)) &&
-      typeof yMinMaxRange?.min === 'number' &&
-      typeof yMinMaxRange?.max === 'number') ||
-      (yAxisVariable?.type === 'date' &&
-        typeof yMinMaxRange?.min === 'string' &&
-        typeof yMinMaxRange?.max === 'string')
-      ? // no padding is made here
-        yMinMaxRange
-      : undefined;
-  }, [data.value, yAxisVariable, categoricalMode]);
+  const xMinMaxDataRange = useMemo(
+    () =>
+      // This useBinning edge case handling is a bit of kludge.
+      // The same situation affects the histogram independent axis (all the time).
+      // The problem is if you make a data request for an x-axis viewport (aka zoom to range), then
+      // the xMin and xMax are calculated from the response, which is just for the requested
+      // viewport.  Using the data-derived xMin and xMax works great for client-side zooming,
+      // because the data doesn't change. But for server-side zooming we can't use xMin and xMax from
+      // the response.
+      // Two possible solutions I can think of:
+      // 1. Client makes a request for unzoomed data to get the xMin and xMax, and remembers this.
+      // 2. Server returns the truncation flags for us, somehow.
+      vizConfig.useBinning
+        ? defaultIndependentAxisRange
+        : data.value != null
+        ? ({ min: data.value.xMin, max: data.value?.xMax } as NumberOrDateRange)
+        : undefined,
+    [data, vizConfig.useBinning]
+  );
+  const yMinMaxDataRange = useMemo(
+    () =>
+      data.value != null
+        ? ({ min: data.value.yMin, max: data.value?.yMax } as NumberOrDateRange)
+        : undefined,
+    [data]
+  );
+
+  const defaultDependentAxisRange = useDefaultAxisRange(
+    yAxisVariable,
+    data.value?.yMin,
+    data.value?.yMinPos,
+    data.value?.yMax,
+    vizConfig.dependentAxisLogScale
```

Comment:
line plot dependent axes are marked by means and medians. For the same reason we disallow log scale on boxplot y and when errorbars are on, should we not allow log scale for y axes on line plot y axis at all?

Reply:
I think your hunch is right, at least for the mean, which is calculated in regular arithmetic space.  The median is based on ranking, so will be the same in log and regular space.  I think we leave this for now.